File: csharp/main/NamespaceDef.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public class NamespaceDef
	{
		public NamespaceDef([NotNull] ProjectDef projectDef, [NotNull] String @namespace)
		{
			Namespace = @namespace;
			Project = projectDef;
			projectDef.Namespaces.Add(Namespace, this);
		}

		[NotNull]
		public String Namespace { get; }

		[NotNull]
		public SortedDictionary<String, FileDef> Files { get; } = new SortedDictionary<String, FileDef>();

		[NotNull]
		public ProjectDef Project { get; }

		[NotNull]
		public Dictionary<String, TypeCustom> DefinedTypes = new Dictionary<String, TypeCustom>();

		/// <inheritdoc />
		public override String ToString()
		{
			return Namespace;
		}
	}
}

File: java/src/main/java/deltix/luminary/NamespaceDef.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;
import java.util.SortedMap;
import java.util.TreeMap;

/**
 * Describes the namespace within the project.
 */
public class NamespaceDef {
    private final String namespace;
    private final ProjectDef project;
    private final SortedMap<String, FileDef> files = new TreeMap<>();
    private final Map<String, TypeCustom> definedTypes = new HashMap<>();

    public NamespaceDef(@NotNull ProjectDef project, @NotNull String namespace) {
        this.project = project;
        this.namespace = namespace;

        project.getNamespaces().put(namespace, this);
    }

    /**
     * String namespace.
     *
     * @return String namespace.
     */
    @NotNull
    public String getNamespace() {
        return namespace;
    }

    /**
     * Project this namespace belongs to.
     *
     * @return Project this namespace belongs to.
     */
    @NotNull
    public ProjectDef getProject() {
        return project;
    }

    /**
     * List of files that contribute to this namespace.
     *
     * @return List of files that contribute to this namespace.
     */
    @NotNull
    public SortedMap<String, FileDef> getFiles() {
        return files;
    }

    /**
     * Map of all types defined within this namespace including nested types.
     *
     * @return Map of all types defined within this namespace.
     */
    @NotNull
    public Map<String, TypeCustom> getDefinedTypes() {
        return definedTypes;
    }
}
